ethereum/core/src/main/java/tech/pegasys/teku/core/signatures/SlashingProtector.java

```diff
@@ -18,50 +18,64 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
 import tech.pegasys.teku.bls.BLSPublicKey;
 import tech.pegasys.teku.core.signatures.record.ValidatorSigningRecord;
+import tech.pegasys.teku.data.slashinginterchange.SlashingProtectionRecord;
+import tech.pegasys.teku.data.slashinginterchange.YamlProvider;
+import tech.pegasys.teku.datastructures.state.ForkInfo;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 
 public class SlashingProtector {
-
+  private static final Logger LOG = LogManager.getLogger();
   private final Map<BLSPublicKey, ValidatorSigningRecord> signingRecords = new HashMap<>();
-
   private final SyncDataAccessor dataAccessor;
+  private final YamlProvider yamlProvider = new YamlProvider();
   private final Path slashingProtectionBaseDir;
 
   public SlashingProtector(
       final SyncDataAccessor dataAccessor, final Path slashingProtectionBaseDir) {
-    this.dataAccessor = dataAccessor;
     this.slashingProtectionBaseDir = slashingProtectionBaseDir;
+    this.dataAccessor = dataAccessor;
   }
 
   public synchronized SafeFuture<Boolean> maySignBlock(
-      final BLSPublicKey validator, final UInt64 slot) {
+      final BLSPublicKey validator, final ForkInfo forkInfo, final UInt64 slot) {
     return SafeFuture.of(
         () -> {
           final ValidatorSigningRecord signingRecord = loadSigningRecord(validator);
-          return handleResult(validator, signingRecord.maySignBlock(slot));
+          return handleResult(
+              validator, forkInfo.getGenesisValidatorsRoot(), signingRecord.maySignBlock(slot));
```

Comment:
It would make more sense to pass genesis validators root into `maySignBlock` so that if it is allowed to sign, it just creates a new `ValidatorSigningRecord` that already holds the right genesis validators root.  It should also refuse to sign if it has a validators root that is different to what is passed in as that indicates that the slashing protection DB is from the wrong chain and something has gone wrong (make sure to log an error message in this case as well as the return or it won't be clear why it refused to sign).

Reply:
we discussed the validation against the existing validators root being a separate ticket?